### PR TITLE
Redirector.ResourceLocation shouldn't return `"", nil`

### PR DESCRIPTION
**Pull request by xudifsd**
https://github.com/GoogleCloudPlatform/kubernetes/issues/3823 I'm not sure if we should log the `"", nil` case, but it doesn't make sense to return this

### CODE REVIEW

**Comment by xudifsd** [comment]
CLA signed

**Comment by bgrant0607** [comment]
@rjnagal I saw you self-assigned #3823. PTAL.

File: pkg/apiserver/interfaces.go [hunk 1]
```diff
@@ -101,5 +101,6 @@ type ResourceWatcher interface {
 // Redirector know how to return a remote resource's location.
 type Redirector interface {
 	// ResourceLocation should return the remote location of the given resource, or an error.
+	// shouldn't return `"", nil`
```
Comment (rjnagal) [inline pkg/apiserver/interfaces.go]: nit: remove the comment. I think its redundant and hard to parse without the context.

**Comment by rjnagal** [comment]
just one nit. LGTM for removing the extra check.

**Comment by xudifsd** [comment]
@rjnagal done

**Comment by rjnagal** [comment]
Thanks @xudifsd Can you squash the two commits into one?

**Comment by xudifsd** [comment]
@rjnagal sure, also rebased to current master

**Comment by rjnagal** [comment]
LGTM

Thanks for your patience. Will merge when travis returns.
